Automation test

yinghuang123 opened the pull request:
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below


# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)


-----------------------------------------------------------------------------------------------------------

jctaillandier committed:
Adding more Analytics rules

v-prasadboke committed:
Update signl4.svg

v-prasadboke committed:
Logo corrected

v-dvedak committed:
Merge pull request #7941 from Azure/v-sabiraj-proofpointtapAdditionallogging

Encoding issue fix for Proofpoint TAP ICM 381702433

v-sabiraj committed:
Fixing the input values for playbooks (#7850)

* Update createSolutionV2.ps1

* Update createSolutionV2.ps1

* Update createSolutionV2.ps1

* Update createSolutionV2.ps1

* Update createSolutionV2.ps1

* Update createSolutionV2.ps1

* Update createSolutionV2.ps1

* Update createSolutionV2.ps1

vakohl committed:
Updating Join kind

v-prasadboke committed:
Update signl4.svg

v-prasadboke committed:
Update signl4.svg

v-prasadboke committed:
Update signl4.svg

v-prasadboke committed:
Update signl4.svg

v-prasadboke committed:
Update signl4.svg

v-prasadboke committed:
Update signl4.svg

v-prasadboke committed:
Update signl4.svg

v-prasadboke committed:
Update signl4.svg

vakohl committed:
EndpointThreatEssetials_AnalyticsRules

vakohl committed:
Update LogonwithExpiredAccount.yaml

v-atulyadav committed:
Merge pull request #7942 from Azure/EndpointThreat_ContentValidation2

EndpointThreatEssetials_AnalyticsRules

sangling committed:
Spelling correction - strucutured

juju4 committed:
docs: add IAM Sentinel Responder to readme

sangling committed:
Added apostrophe to hasn't

oshezaf committed:
Update ASIM_GetWindowsUserType.yaml

oshezaf committed:
Check for SID S-1-0-0

Commit:
[ASIM Parsers] Generate deployable ARM templates from KQL function YAML files.

oshezaf committed:
Update ASIM_GetUserType calls

oshezaf committed:
Merge branch 'asim/update-windows-usertype-function' of https://github.com/Azure/Azure-Sentinel into asim/update-windows-usertype-function

Commit:
[ASIM Parsers] Generate deployable ARM templates from KQL function YAML files.

oshezaf committed:
Version update

oshezaf committed:
Merge branch 'asim/update-windows-usertype-function' of https://github.com/Azure/Azure-Sentinel into asim/update-windows-usertype-function

vakohl committed:
InfobloxParser_SrcPortNumber_Issue

Commit:
Merge remote-tracking branch 'origin/master' into ASIM_DNS_Infoblox_Parser

Commit:
[ASIM Parsers] Generate deployable ARM templates from KQL function YAML files.

vakohl committed:
removing unwanted changes

vakohl committed:
Merge branch 'ASIM_DNS_Infoblox_Parser' of https://github.com/Azure/Azure-Sentinel into ASIM_DNS_Infoblox_Parser

preetikr committed:
Update ReadMe.md

AcceleryntSecurityDev committed:
Delete Playbooks/Edgescan-AzureSentinel-Integration directory

AcceleryntSecurityDev committed:
Create init

lem0w committed:
Sign In Anomaly Query

Initial file creation.

lem0w committed:
Rename SignInAnomaly to SignInAnomaly.yaml

adding file extension

AcceleryntSecurityDev committed:
Create README.md

AcceleryntSecurityDev committed:
Add files via upload

AcceleryntSecurityDev committed:
Delete Edgescan_Integration_API_Token_2.png

AcceleryntSecurityDev committed:
Delete init

AcceleryntSecurityDev committed:
Add files via upload

AcceleryntSecurityDev committed:
Add files via upload

AcceleryntSecurityDev committed:
Update README.md

vakohl committed:
Update Anomalous Sign-in Activity.yaml

v-dvedak committed:
Merge pull request #7945 from sangling/patch-1

Spelling correction - strucutured

oshezaf committed:
Merge pull request #7947 from Azure/asim/update-windows-usertype-function

Asim/update windows usertype function

v-dvedak committed:
Merge pull request #7946 from sangling/patch-2

Added apostrophe to hasn't

v-dvedak committed:
Merge pull request #7927 from Azure/UEBA_Essentials_ContentTesting3

UEBA_Essentials_HuntingQuery

vakohl committed:
Infoblox_Sample_Data_Update

vakohl committed:
Merge pull request #7939 from Azure/InfobloxParser_DNSClient

Infoblox parser dns client

spatelte committed:
Check T.io status and update in storage (#7502)

* Check tio status and update in storage

* remove status check in send chunk

* Changes validation check export job

* log changes according the export job status

* Change condition and logs

* Change job status to Job_details

* update TenableIOAzureSentinelConnector.zip

---------

Co-authored-by: Vara Prasad Onteru <varap11@gmail.com>

dependabot[bot] committed:
Bump flat and mocha (#7007)

* Bump flat and mocha

Bumps [flat](https://github.com/hughsk/flat) to 5.0.2 and updates ancestor dependency [mocha](https://github.com/mochajs/mocha). These dependencies need to be updated together.


Updates `flat` from 4.1.0 to 5.0.2
- [Release notes](https://github.com/hughsk/flat/releases)
- [Commits](https://github.com/hughsk/flat/compare/4.1.0...5.0.2)

Updates `mocha` from 7.1.1 to 10.2.0
- [Release notes](https://github.com/mochajs/mocha/releases)
- [Changelog](https://github.com/mochajs/mocha/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mochajs/mocha/compare/v7.1.1...v10.2.0)

---
updated-dependencies:
- dependency-name: flat
  dependency-type: indirect
- dependency-name: mocha
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update text from Azure Sentinel to Microsoft Sentinel

* Update package.json

* update versions

* Revert "update versions"

This reverts commit d9968e10b49ec9b10a9c4e1ab2ef2c59788dba29.

* Update package.json

* Update package-lock.json

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: v-atulyadav <104008048+v-atulyadav@users.noreply.github.com>
Co-authored-by: v-rusraut <v-rusraut@microsoft.com>

v-prasadboke committed:
api version update

TomerAligaev committed:
add forcePoint imNetworkSessionBuiltin pasrsers list

oshezaf committed:
Merge pull request #7954 from Azure/Update_Infoblox_Sample_data

Infoblox_Sample_Data_Update

oshezaf committed:
Merge pull request #7955 from Azure/feature/taligaev/update_imNetworkSession

add forcePoint imNetworkSessionBuiltin pasrsers list

oshezaf committed:
Merge pull request #7949 from Azure/ASIM_DNS_Infoblox_Parser

Asim dns infoblox parser update

v-dvedak committed:
Merge pull request #7934 from Azure/v-prasadboke-signl4-logo

Signl4 Logo Correction

v-dvedak committed:
Merge pull request #7789 from austinmccollum/patch-1

update instructions for codeless connector

v-atulyadav committed:
Update stats.md (#7956)

lem0w committed:
Updating query based on recent comment

Adding Score and Baseline as part of the series_decompose_anomalies, as well adding the kind:schedule and custom Details.

v-vdixit committed:
Palo Alto CDL CreateUI update

v-mchatla committed:
Create GCPAuditLogs.json

v-dvedak committed:
Merge pull request #7928 from gh-andrem/patch-1

Update Endpoint Agent Health Status Report.yaml

v-dvedak committed:
Merge pull request #7960 from Azure/v-mchatla/GCPAuditLogs-Dataconnector

Adding Dataconnector folder to GoogleCloudPlatformAuditLogs Solution

deggis committed:
ASimTester: Relax NetworkSession/EventSeverity to Recommended

In current ASIM common fields and NetworkSession schemas
EventSeverity is listed as a Recommended field.

Change its class from Mandatory to Recommended.

https://learn.microsoft.com/en-us/azure/sentinel/normalization-common-fields
https://learn.microsoft.com/en-us/azure/sentinel/normalization-schema-network

AcceleryntSecurityDev committed:
Create azuredeploy.json

AcceleryntSecurityDev committed:
Delete azuredeploy1.json

AcceleryntSecurityDev committed:
Create azuredeploy.json

AcceleryntSecurityDev committed:
Delete azuredeploy2.json

AcceleryntSecurityDev committed:
Create azuredeploy.json

AcceleryntSecurityDev committed:
Delete azuredeploy3.json

AcceleryntSecurityDev committed:
Update README.md

daspiker committed:
updating solutionmetadata.json file

v-vdixit committed:
updating createui

v-dvedak committed:
Merge pull request #7896 from juju4/devel-AzureResourceGraph-Incident

feat: add Playbooks/Enrich-AzureResourceGraph-Incident

v-dvedak committed:
Merge pull request #7893 from Accelerynt-Security/04-21-2023--AS-Azure-AD-Enable-User

04 21 2023  as azure ad enable user

v-dvedak committed:
Merge pull request #7891 from Accelerynt-Security/04-13-2023--AS-Incident-Spiderfoot-Scan

04 13 2023  as incident spiderfoot scan

v-vdixit committed:
Updating versions

v-vdixit committed:
Merge branch 'master' into v-vdixit/KQL-validation-test-hunting-queries

v-vdixit committed:
Update skipvalidations

v-dvedak committed:
Merge pull request #7778 from Azure/v-prasadboke-MicrosoftDefenderThreatIntelligence-Repackaging

Microsoft Defender Threat Intelligence Repackaging

v-vdixit committed:
detections renamed

v-vdixit committed:
Hunting queries renamed

v-dvedak committed:
Updated the broken links (#7957)

v-dvedak committed:
Merge pull request #7959 from Azure/v-vdixit/updating-PaloAlto-CDL

Palo Alto CDL CreateUI update

v-vdixit committed:
Solution repackaging

daspiker committed:
added Solution_Island.json

v-vdixit committed:
updating detections version and links

daspiker committed:
removed workbookbladedescription

daspiker committed:
updated package

lem0w committed:
Corrected query logic

Updated the query logic to exclude events from known networks adding a join with sign-in to enrich the result it will now look for sign in events from an IP flagged as an anomaly where it contains a correct password

lem0w committed:
Fixed indentation for query

Fixed indentation for query

v-vdixit committed:
Revert "updating detections version and links"

This reverts commit 1d3449488b38ba1b1e6f5a58a1619fe2cff2cb21.

v-vdixit committed:
Revert "Solution repackaging"

This reverts commit fb37b28fa10045ac85519397ae3fe8d1d716791c.

v-vdixit committed:
Updating version

v-mchatla committed:
Analytic Rule Validation Fix

Analytic Rule Validation Fix

v-vdixit committed:
updating Dev-0586

v-prasadboke committed:
Package update

v-sabiraj committed:
Rectifying the api version for templatespecs

v-vdixit committed:
Updating analytic rule

v-prasadboke committed:
Repackaging Network Threat Essential

v-dvedak committed:
Merge pull request #7969 from Azure/v-sabiraj-fixingAPIversionsfortemplatespecs

Rectifying the api version for templatespecs

javiersoriano committed:
Fix UEBA deployment

Adding BehaviorAnalyticsInsights OMS solution deployment

v-sabiraj committed:
Changing parser name

oshezaf committed:
asim/update-win-parsers-with-usertype

oshezaf committed:
Merge pull request #7961 from deggis/asim/asimtester_schema_fix

ASimTester: Relax NetworkSession/EventSeverity to Recommended

v-sabiraj committed:
updating package

Commit:
Merge remote-tracking branch 'origin/master' into asim/update-win-parsers-with-usertype

Commit:
[ASIM Parsers] Generate deployable ARM templates from KQL function YAML files.

daspiker committed:
updated apiVersion to address arm-ttk issue

v-vdixit committed:
Repackaging Cisco ISE

v-vdixit committed:
Merge branch 'v-vdixit-UpdatingParsernameforciscoise' of https://github.com/Azure/Azure-Sentinel into v-vdixit-UpdatingParsernameforciscoise

v-vdixit committed:
Updating zip

v-vdixit committed:
updating UI

sreedharande committed:
Merge pull request #7971 from javiersoriano/patch-4

Fix UEBA deployment

yinghuang123 committed:
add error_handler within the scripts.

tj-senserva committed:
Fixing broken link

tj-senserva committed:
Fixing broken link

tj-senserva committed:
Fixing broken link

v-dvedak committed:
Merge pull request #7797 from austinmccollum/patch-3

update GUID creation for uniqueness

v-rusraut committed:
Repackaging - Zinc Open Source

v-dvedak committed:
Merge pull request #7940 from daspiker/IslandSentinelPackage

adding initial package

v-dvedak committed:
Merge pull request #7965 from Azure/v-vdixit/Legacy-IOC-rename

Legacy IOC repackaging with content names update

v-prasadboke committed:
ui update

v-dvedak committed:
Merge pull request #7970 from Azure/v-prasadboke-NetworkProtectionEssentials

Repackaging Network Threat Essential

v-vdixit committed:
updating solution metadata

v-prasadboke committed:
description update

pemontto committed:
🐛 Fix incorrect column name

tj-senserva committed:
Updating API Versions

PCNZ committed:
Update TenableIO.json

Remove note about "can't mix Windows and Linux apps in the same region"
This limitation no longer exists in Azure
https://learn.microsoft.com/en-us/azure/app-service/overview#limitations

NikTripathi committed:
CiscoSEG Parser Optimization Regarding IcM 355468251 (#7503)

v-dvedak committed:
Merge pull request #7973 from Azure/v-vdixit-UpdatingParsernameforciscoise

Repackaging Cisco ISE with parser name update

oshezaf committed:
Merge pull request #7982 from pemontto/entity-column-typo

🐛 Fix incorrect column name

v-dvedak committed:
Merge pull request #7952 from lem0w/lem0w-SignIn-Detection

Sign In Anomaly Query

oshezaf committed:
Update vimFileEventM365D.yaml

oshezaf committed:
Merge branch 'asim/update-win-parsers-with-usertype' of https://github.com/Azure/Azure-Sentinel into asim/update-win-parsers-with-usertype

Commit:
Merge remote-tracking branch 'origin/master' into asim/update-win-parsers-with-usertype

Commit:
[ASIM Parsers] Generate deployable ARM templates from KQL function YAML files.

v-sabiraj committed:
Merge branch 'master' into v-sabiraj-tiupdate

v-sabiraj committed:
Updated package to fix the api version

oshezaf committed:
Add ATT&CK and remediation fields

oshezaf committed:
Merge pull request #7972 from Azure/asim/update-win-parsers-with-usertype

asim/update-win-parsers-with-usertype

oshezaf committed:
Merge pull request #7881 from Azure/asim/notification-schema

Create ASimNotification.yaml

v-dvedak committed:
Merge pull request #7301 from Azure/v-vdixit/KQL-validation-test-hunting-queries

KQL Validations for Multiple data sources hunting queries

daspiker committed:
updated package with apiVersion fix

juju4 committed:
feat: add Playbooks/AzureMonitor-ManagedId (#7894)

* feat: add Playbooks/AzureMonitor-ManagedId

* fix: add metadata

* docs: add mention of liquid map name

v-atulyadav committed:
Repackaged Microsoft 365, MicrosoftDefenderForEndpoint, Windows Forwarded Events (#7978)

* Repackaged

* update rule query

v-vdixit committed:
Standalone Content Renaming (#7981)

yinghuang123 committed:
refactor the automation tree structure, improved the error handling.

v-dvedak committed:
Merge pull request #7882 from Azure/v-sabiraj-tiupdate

Updated TI solution to update Analytic rules

v-rusraut committed:
Repackaging - UEBAEssentials

v-rusraut committed:
Update mainTemplate.json

v-prasadboke committed:
Endpoint threat Protection

v-prasadboke committed:
update

v-dvedak committed:
Merge pull request #7977 from Azure/v-rusraut/zincopensource

Repackaging - Zinc Open Source

v-rbajaj committed:
Repackaging Cloud Identity Threat Protection and Cloud Service Threat Protection

v-atulyadav committed:
Update SolutionMetadata.json

v-atulyadav committed:
Update 2.0.1.zip

v-rusraut committed:
update MainTemplate

v-rusraut committed:
Merge branch 'v-rusraut/UEBAEssentials' of https://github.com/Azure/Azure-Sentinel into v-rusraut/UEBAEssentials

TomerAligaev committed:
Add back GetUserType (#7994)

* Add back GetUserType

* fix description and calling in query

v-rbajaj committed:
Repackaging Azure Activity

daspiker committed:
removed hidden files from package

daspiker committed:
updated Solution_Island.json

tj-senserva committed:
Delete bad version

tj-senserva committed:
Replace template

tj-senserva committed:
Merge branch 'Azure:master' into master

v-mchatla committed:
Updated Alert Description

kishan-coh commented on Solutions/CohesitySecurity/Tools/assign_sentinel_role.sh:
```diff
@@ -0,0 +1,22 @@
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+#
+# Description: This script assigns the "Microsoft Sentinel Playbook Operator" role
```
Please move the description to the top of the file as we do for all other files.

Please note that this is a generic comment for all the new files. I don't want to overload the same comments across all the files.

yinghuang123 replied:
done

kishan-coh commented on Solutions/CohesitySecurity/Tools/assign_sentinel_role.sh:
```diff
@@ -0,0 +1,22 @@
+SCOPE="/subscriptions/$subscription_id"
+
+# Set the subscription
+az account set --subscription "$subscription_id"
```
I see that this script expects subscription_id and client_id to be passed. Could you please add the validation at the start related to these parameters(that they are actually present). Please note that these scripts should work independently as well, so it should be easy for a user to look at the description and run the script and when the script is run, these basic input validation is necessary.


Please note that this is a generic comment for all the new files. I don't want to overload the same comments across all the files.

yinghuang123 replied:
No, it seems there might be a misunderstanding regarding how shell scripts work. The parameters in question are not passed in. Instead, they are global variables set here:
    Solutions/CohesitySecurity/Tools/sentinel_provisioning.sh:12 . ./json_parser.sh

This is one reason why I prefer using shell scripts for installation and configuration tasks.

It's not always the case that every script should work independently. If you want to execute assign_sentinel_role.sh, you'll need to manually add the following line:
    . ./json_parser.sh

This ensures that the necessary variables are set and available for the script to function correctly.

kishan-coh replied:
1. "It's not always the case that every script should work independently."
> In all the scripts that you have added, I don't see a reason for them not to work independently. It's lost opportunity.

2. I think we chose a bad methodology. Having global variables when we are dealing with 10s of scripts is bad. Now every one has to make sure the same name is not used as a local variable. I don't see why you have chosen that path.
 
3. Why does it matter whether the variables are global or local to do validation? The script is expected to do stepA, stepB and they are dependent on variableA and variableB to be set. My question is why is the existence of variableA and variableB not being set. Is that a basic validation before actually going and doing the operations?

kishan-coh commented on Solutions/CohesitySecurity/Tools/assign_sentinel_role.sh:
```diff
@@ -0,0 +1,22 @@
+
+# Set the subscription
+az account set --subscription "$subscription_id"
+error_handler "Failed to set the subscription."
```
Adding this error_handler is good. However, I have seen scripts usually add some flags on the top to [stop proceeding](https://stackoverflow.com/questions/2870992/automatic-exit-from-bash-shell-script-on-error) when an error is encountered. Consider adding such flags as well.

yinghuang123 replied:
Adding `set -e` at the top of the script will cause the script to exit immediately if any command returns a non-zero status. If I want my script to be more robust and fail fast in case of any error, adding `set -e` can be a good idea. However, if I want to handle errors more gracefully with custom error messages, I can use the `error_handler` function in my script, as I am already doing.

So, if I want to maintain the current error handling approach with custom error messages, I don't need to add `set -e`. But if I prefer a fail-fast approach, I can add `set -e` at the top of my script.

yinghuang123 replied:
I don't want to add `set -e`, as I need custom error messages for debugging purposes. Using custom error messages allows me to provide more specific and helpful information for each step of my script, which is useful for identifying issues and understanding the context of the error.

kishan-coh replied:
1. The problem comes when someone adds a new command and forgets to add error_handler.
2. I looked at many error messages that are added to the error handler. All of them a single line static strings. None of them have any 'variables' added. I don't understand what additional information they're providing. When 'set -e' is set, it show the exact line where the error occurred and the error handler is not providing any additional information. In fact, I would argue that it will make it difficult to debug if two error messages are same, we don't know which command produced the error.

kishan-coh commented on Solutions/CohesitySecurity/Tools/assign_sentinel_role.sh:
```diff
@@ -0,0 +1,22 @@
+error_handler "Failed to set the subscription."
+
+# Assign the role
+az role assignment create --role "$ROLE" --assignee "$client_id" --scope "$SCOPE"
```
I see that 'client_id' is being assigned the Playbook operator role here. But, the client we create doesn't deal with playbooks. It only deals with function apps. Please remove this file if it's not necessary.


If you end up keeping this file, please rename it to assign_sentinel_playbook_role.sh

yinghuang123 replied:
done
R Solutions/CohesitySecurity/Tools/assign_sentinel_role.sh -> Solutions/CohesitySecurity/Tools/assign_sentinel_playbook_operator_role.sh

kishan-coh commented on Solutions/CohesitySecurity/Tools/config.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/zsh
```
Please add a description to the file. Please do it in all the new files which don't have a description.

yinghuang123 replied:
done

kishan-coh commented on Solutions/CohesitySecurity/Tools/grant_role.sh:
```diff
@@ -0,0 +1,26 @@
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+# Description: This script assigns the "Microsoft Sentinel Contributor" role to a client in the specified workspace.
```
Please rename the file to grant_sentinel_contributor_role.sh

yinghuang123 replied:
done

kishan-coh commented on Solutions/CohesitySecurity/Tools/grant_role.sh:
```diff
@@ -0,0 +1,26 @@
+error_handler "Failed to get the workspace scope."
+
+# Assign the role to the client
+az role assignment create --assignee-object-id "$object_id" --role "$ROLE" --scope "$SCOPE"
```
What is object_id here? I see a new field added in cohesity.json. The name is too generic. Please name it to suit the purpose.

Also, if my understanding is correct, in this command, the 'object_id' is a unique UUID that should be created by us. Why is not generated  dynamically here and being sent as part of cohesity.json?

yinghuang123 replied:
no, we could not replace it with run time uuid.
The `object_id` in the command `az role assignment create --assignee-object-id "$object_id" --role "$ROLE" --scope "$SCOPE"` refers to the unique identifier (ID) of the Azure Active Directory (Azure AD) principal (user, group, or service principal) to which the specified role should be assigned.

When you create a role assignment using Azure CLI, you need to provide the object ID of the Azure AD principal that you want to assign the role to. The role assignment then grants the specified role's permissions to that principal within the defined scope (such as a resource group, a subscription, or a specific resource).

kishan-coh commented on Solutions/CohesitySecurity/Tools/deploys.sh:
```diff
@@ -0,0 +1,13 @@
+#!/bin/zsh
```
Consider renaming this to deploy_solution.sh

yinghuang123 replied:
done

kishan-coh commented on Solutions/CohesitySecurity/Tools/deploy_playbooks.sh:
```diff
@@ -0,0 +1,35 @@
+            --name "${playbook_name}_Deployment" \
+            --resource-group "$resource_group" \
+            --template-file "$template_file" \
+            --parameters PlaybookName="$playbook_name"
```
Consider adding 'Auto_' in front of the playbook name to easily isolate the automation playbooks vs the actual playbooks.

yinghuang123 replied:
no....then the py test code also need to change to test the Auto_* playbook...
I think the py test should always test against the env as close as possible to real one.

kishan-coh replied:
I don't understand how naming a playbook has anything to do with 'env as close as possible to real one'. The real deployment does give the user the option to provide a name for the playbook.

The reason for asking this is that it's getting confusing on the UI to see all the playbooks having the same name and it's getting difficult to choose the right one on the incidents.

kishan-coh commented on Solutions/CohesitySecurity/Tools/configure_permissions.sh:
```diff
@@ -0,0 +1,67 @@
+            error_handler "Failed to send the batch request."
+
+            echo "Permissions configured for the playbook."
+        else
```
Please correct the indentation of 'else'

yinghuang123 replied:
done

kishan-coh commented on Solutions/CohesitySecurity/Tools/configure_permissions.sh:
```diff
@@ -0,0 +1,67 @@
+cd "$SCRIPTPATH"
+#
+# Description: This script assigns Microsoft Sentinel Automation Contributor role
+# permissions to the specified app display name and role name.
```
"to the specified app display name and role name."

- But these are already hardcoded below. Please fix the description.

yinghuang123 replied:
done

kishan-coh commented on Solutions/CohesitySecurity/Tools/configure_permissions.sh:
```diff
@@ -0,0 +1,67 @@
+    az rest --method POST \
+        --url "https://management.azure.com/batch?api-version=2020-06-01" \
+        --headers "Content-Type=application/json" \
+        --body "{\"requests\":[{\"content\": $request_body, \"httpMethod\":\"PUT\", \"name\":\"ed804683-92f0-4609-a3a2-132de30860e9\", \"url\":\"https://management.azure.com/subscriptions/$subscription_id/resourceGroups/$resource_group/providers/Microsoft.Authorization/roleAssignments/$role_assignment_id?api-version=$api_version\"}]}"
```
What's 'ed804683-92f0-4609-a3a2-132de30860e9'? And why is it hardcoded?

yinghuang123 replied:
just replaced it with a run time uuid value.

kishan-coh commented on Solutions/CohesitySecurity/Tools/configure_permissions.sh:
```diff
@@ -0,0 +1,67 @@
+# Description: This script assigns Microsoft Sentinel Automation Contributor role
+# permissions to the specified app display name and role name.
+# It first retrieves the principal ID and role definition ID, then
+# grants the necessary permissions for the playbook.
```
"grants the necessary permissions for the playbook". What playbook are we talking about? I don't see any variable in the script which refers to the playbook name/id?

yinghuang123 replied:
it is in the workspace setting page, to give Microsoft Sentinel permissions to resource groups that contain the playbooks

kishan-coh replied:
But that is what configure_permissions.sh is doing.

yinghuang123 replied:
then what's your suggestion for this description?

kishan-coh commented on Solutions/CohesitySecurity/Tools/get_role_definition_id.sh:
```diff
@@ -0,0 +1,27 @@
+role_definitions=$(az role definition list --output json)
+
+# Filter role definition by roleName and extract the id
+filtered_role_definition_id=$(echo $role_definitions | jq -r ".[] | select(.roleName == \"$role_name\") | .name")
```
Fyi, all the 'az' command by default have an in-built jmespath available to filter based any attributes. So, technically, the role_definitions and filtered_role_definition_id be combined.

yinghuang123 replied:
just updated, will test again later, to see whether it works.

kishan-coh commented on Solutions/CohesitySecurity/Tools/configure_permissions.sh:
```diff
@@ -0,0 +1,67 @@
+    )
+
+    # Send a batch request
+    az rest --method POST \
```
I'm trying to find the difference between this file and the grant_role.sh. My understanding is that both of these scripts are doing the same thing. Could you please help me explain the difference?


Btw, we should directly post the request to https://management.azure.com/subscriptions/$subscription_id/resourceGroups/$resource_group/providers/Microsoft.Authorization/roleAssignments/$role_assignment_id?api-version=$api_version

I don't see a reason for using 'https://management.azure.com/batch' here.

yinghuang123 replied:
they are different.
grant_role.sh is assign role to user. And it is a done through az cli. Which means, it should be supported by MS for a long time. 
configure_permissions.sh is to add playbook permission to workspace. This way is like hack.
I got https://management.azure.com/batch through chrome console, and mimic the similar request, that's why I said it is a hack.

what's your reason for using https://management.azure.com/subscriptions/$subscription_id/resourceGroups/$resource_group/providers/Microsoft.Authorization/roleAssignments/$role_assignment_id?api-version=$api_version ?

kishan-coh replied:
It is no longer a hack if we remove the 'batch' REST API and directly use the role_assignment REST API. The reason for using the API I mentioned directly is that 'batch' is not a 'documented' REST API call, but the roleAssignments REST API is a documented and support REST API call.

I'm not sure if you have got a chance to explore, but there may just be an az cli command to perform this just as you have done in grant_role.sh.

yinghuang123 replied:
Please note that this process does not involve role assignment. Instead, it follows the manual configuration workflow found in the Azure Portal under Microsoft Sentinel Workspace > Settings > Playbook permissions.

Despite extensive research, I have been unable to find a corresponding Azure CLI command for this specific task. It is possible that a relevant Azure CLI command was available in the past but has since been removed. As a result, some older posts may reference this functionality, but none of the suggested solutions are currently functional.

kishan-coh replied:
"Please note that this process does not involve role assignment"
> Why do you say that? If you look at the REST API, it doing exactly that. Look for the string 'roleAssignments'


Also, you didn't answer regarding the use of the official support REST API. I provided the URL.

kishan-coh commented on Solutions/CohesitySecurity/Tools/grant_keyvault_permission.sh:
```diff
@@ -0,0 +1,42 @@
+
+# Set variables
+keyvault_prefix="cohesitypro"
+playbook_names=("Cohesity_Close_Helios_Incident" "Cohesity_CreateOrUpdate_ServiceNow_Incident" "Cohesity_Delete_Incident_Blobs" "Cohesity_Restore_From_Last_Snapshot" "Cohesity_Send_Incident_Email")
```
Key vault permission should only be necessary for Cohesity_Close_Helios_Incident and Cohesity_Restore_From_Last_Snapshot

yinghuang123 replied:
done

kishan-coh commented on Solutions/CohesitySecurity/Tools/grant_keyvault_permission.sh:
```diff
@@ -0,0 +1,42 @@
+# Set variables
+keyvault_prefix="cohesitypro"
+playbook_names=("Cohesity_Close_Helios_Incident" "Cohesity_CreateOrUpdate_ServiceNow_Incident" "Cohesity_Delete_Incident_Blobs" "Cohesity_Restore_From_Last_Snapshot" "Cohesity_Send_Incident_Email")
+user_object_id="142355ec-a29b-40b7-81c3-e769c76b1756"
```
What's user_object_id and why is it hardcoded?

yinghuang123 replied:
we have to introduce some hardcode, bellowing is the user info for 142355ec:
{
    "businessPhones": [],
    "displayName": "Ravi Varanasi",
    "givenName": "Ravi",
    "id": "142355ec-a29b-40b7-81c3-e769c76b1756",
    "jobTitle": null,
    "mail": null,
    "mobilePhone": null,
    "officeLocation": null,
    "preferredLanguage": "en",
    "surname": "Varanasi",
    "userPrincipalName": "cohesity-siem_outlook.com#EXT#@cohesitysiemoutlook.onmicrosoft.com"
}
just moved the user_object_id into cohesity.json file

kishan-coh replied:
1. Can't we get this information from an 'az cli' command?
2. This automation should be usable from any account. For example, I don't use Ravi's account.
3. Why does a user need the 'Get' privilege on Key vault? Could you please mention the workflow that requires the user to have this privilege? And the documentation in our code which says that the user requires this privelege?

kishan-coh replied:
https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-rest

https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-cli

yinghuang123 replied:
Yes or no, to obtain this information from the CLI, we need to search using either "Ravi Varanasi" or "cohesity-siem_outlook.com#EXT#@cohesitysiemoutlook.onmicrosoft.com". However, we still require some hardcoded input for the search. Among these options, the ID is more likely to be unique for identifying a user.

If you decide to run the automation from a different account, simply update the cohesity.json file to use the other user's ID.

kishan-coh commented on Solutions/CohesitySecurity/Tools/grant_keyvault_permission.sh:
```diff
@@ -0,0 +1,42 @@
+# Description: This script grants "Get" permissions to the specified user and playbooks on the Key Vault.
+
+# Set variables
+keyvault_prefix="cohesitypro"
```
I though we already have a config variable for the prefix. Please reuse it.

yinghuang123 replied:
done

kishan-coh commented on Solutions/CohesitySecurity/Tools/setup.sh:
```diff
@@ -0,0 +1,22 @@
+. ./config.sh
+error_handler "Failed to configure resources."
+
+. ./test.sh
```
A 'setup.sh' should not include test.sh. Please rename the file appropriately or remove this invocation here and create a new file called run.sh which calls both setup.sh and test.sh

yinghuang123 replied:
just renamed to sentinel_provisioning.sh

kishan-coh commented on Solutions/CohesitySecurity/cohesity.json:
```diff
@@ -1,22 +1,28 @@
 {
     "api_key": "<cohesity-helios-api-key>",
     "resource_group": "<resource-group-name>",
+    "resource_type": "Microsoft.OperationalInsights/workspaces",
```
I don't see resource_type being used anywhere. Please remove it and add it a future PR when necessary.

yinghuang123 replied:
yes, not in use now, removed.

kishan-coh commented on Solutions/CohesitySecurity/cohesity.json:
```diff
@@ -1,22 +1,28 @@
 {
     "api_key": "<cohesity-helios-api-key>",
     "resource_group": "<resource-group-name>",
```
We're technically not honoring 'resource_group and 'workspace_name' as we are just using these values as prefix to create a unique a unique rg and workspace. Please rename the config variable names to say resource_group_prefix and workspace_name_prefix.

yinghuang123 replied:
no, you are incorrect.
Read this file:
    Solutions/CohesitySecurity/Tools/json_parser.sh
If you specify resource_group, then it will use the value.
if not, then it will generate a unique one, so as to have a clean environment to install/configure/test

kishan-coh replied:
Ack.

Could you tell me what happens to the automation if the resource group name that already exists is provided. Do we validate and fail fast saying that that the RG is already present, so we can't do automation?

kishan-coh commented on Solutions/CohesitySecurity/cohesity.json:
```diff
@@ -1,22 +1,28 @@
     "user_email": "********",
     "resource_url": "https://management.azure.com/",
     "azure_url_scope": "https://management.azure.com/.default",
     "workspace_id": "********",
```
Is the workspace_id required anywhere?

yinghuang123 replied:
not needed anymore, deleted.

kishan-coh commented on Solutions/CohesitySecurity/cohesity.json:
```diff
@@ -1,22 +1,28 @@
     "blob_container_name": "cohesity-extra-parameters",
-    "workspace_name": "<workspace-name>"
+    "workspace_name": "<workspace-name>",
+    "service_now_username": "********",
```
I have not seen any of the service_now parameters used in this PR. Please add them in a future PR when it's used.

yinghuang123 replied:
they are used in test.

yinghuang123 replied:
Solutions/CohesitySecurity/Tests/az.test.py:46             self.service_now_username = config["service_now_username"]

v-dvedak committed:
Merge pull request #7967 from Azure/v-mchatla/AzureActiveDirectory-AnalyticRuleFix

Azure Active Directory - Analytic Rule Validation Fix

kishan-coh commented on Solutions/CohesitySecurity/Tools/setup.sh:
```diff
@@ -0,0 +1,22 @@
+
+. ./test.sh
+error_handler "Failed to run tests."
+
```
After the test is completed, please delete the resource group.

yinghuang123 replied:
move this to the last.
as still great efforts to have all the test passed.

kishan-coh replied:
Why not just add it and it will automatically delete when all the tests pass instead of punting it to the future? This also provides the user a way to delete the resource group from command line in case a test fails.

v-dvedak committed:
Merge pull request #7905 from Lodewyk-Git/patch-5

Update so rule can deploy from template

ajkallur committed:
Adding 3 New Playbooks to Defender Threat Intelligence solution (#7987)

* Adding 3 New Playbooks to Defender Threat Intelligence solution

Adding 3 New Playbooks to Defender Threat Intelligence solution

MDTI-PassiveDNS
MDTI-PassiveDNSReverse
MDTI-Trackers

* 3 spelling corrections in comments recommended by the reviewer

3 spelling corrections in comments recommended by the reviewer

v-atulyadav committed:
Repackage AADIP

v-dvedak committed:
Merge pull request #8015 from Azure/v-atulyadav/aadip

Repackage AADIP

MeirLevinMicrosoft committed:
AMA troubleshoot v2 (#7966)

* Init

* Init

* Refactor cef_AMA_troubleshoot.py

* Add new args

* Arrange getargs

* PR changes 1

* fix decoding

* Done decoding fixes

* fix for python2

* Add PY version for decoding

* PR fixes

* DEFAULT_MACHINE_ENV to global

v-prasadboke committed:
Workbook update

v-atulyadav committed:
Update 2.0.2.zip

dependabot[bot] committed:
Bump snowflake-connector-python in /Solutions/Snowflake/Data Connectors (#7944)

Bumps [snowflake-connector-python](https://github.com/snowflakedb/snowflake-connector-python) from 2.6.1 to 2.8.3.
- [Release notes](https://github.com/snowflakedb/snowflake-connector-python/releases)
- [Commits](https://github.com/snowflakedb/snowflake-connector-python/compare/v2.6.1...v2.8.3)

---
updated-dependencies:
- dependency-name: snowflake-connector-python
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

v-sabiraj committed:
Update __init__.py for Jira (#7915)

* Update __init__.py

* Updated the zip package

Updated the zip package

* Added logging in else case

Added logging in else case

---------

Co-authored-by: Meena Kumari Chatla <v-mchatla@microsoft.com>

v-dvedak committed:
Merge pull request #7976 from tj-senserva/master

Fixing broken link

v-prasadboke committed:
Repackaging Microsoft 365 Defender

v-rbajaj committed:
Updated workbook description

javiersoriano committed:
Updating alert rule API versions

updating API versions to account for MITRE tactics and techniques

v-rusraut committed:
Repackaging - Windows Security Events

v-dvedak committed:
Merge pull request #7968 from Azure/v-prasadboke-digitalshadows

Package update for Digital Shadows

v-dvedak committed:
Merge pull request #8007 from Azure/v-rbajaj/RepackagingAzureActivity

Repackaging Azure Activity

v-sabiraj committed:
Updating confluence function to add exception

daspiker committed:
removed REST comment

v-sabiraj committed:
Updated instruction steps for GitHub Connector

yinghuang123 committed:
some update according to review.

yinghuang123 committed:
manual update to cohesity.json.

v-dvedak committed:
Merge pull request #8026 from Azure/v-sabiraj-addingexceptionforConfluence

Updating confluence function to add exception

v-sabiraj committed:
Fixing arm ttk validations

v-rusraut committed:
Update 2.0.4.zip

v-dvedak committed:
Merge pull request #8024 from Azure/v-rusraut/windowssecuevent

Repackaging - Windows Security Events

v-sabiraj committed:
Updating description

daspiker committed:
added 2.0.0.zip back

sreedharande committed:
Merge pull request #8023 from javiersoriano/patch-5

Updating alert rule API versions

yinghuang123 committed:
Merge remote-tracking branch 'origin/master' into CohesitySecurity.internal

ntwrite committed:
Update IPEntity_imNetworkSession.yaml

Typo

oshezaf committed:
Update README.md

v-sabiraj committed:
Updated API versions for Darktrace Solution

v-atulyadav committed:
Update IPEntity_imNetworkSession.yaml

v-prasadboke committed:
create ui

v-dvedak committed:
Merge pull request #8036 from Azure/v-sabiraj-UpdatingDarktrace

Updated API versions for Darktrace Solution

v-dvedak committed:
Merge pull request #8035 from ntwrite/patch-1

Update IPEntity_imNetworkSession.yaml

v-sabiraj committed:
Updated NIST Solution for workbook

v-dvedak committed:
Merge pull request #7995 from Azure/v-prasadboke-Endpoint/MicrosoftDefender

Endpoint threat Protection & Microsoft Defender for Cloud Repackaging

v-dvedak committed:
Merge pull request #8020 from Azure/v-prasadboke-microsoft365/azureactivity

Repackaging Microsoft 365 Defender

v-dvedak committed:
Merge pull request #8027 from Azure/v-sabiraj-UpdatinginstructionstepsforGitHubconnector

Updated instruction steps for GitHub Connector

v-rbajaj committed:
Merge branch 'master' into v-rbajaj/RepackagingSolutionsKQL

v-dvedak committed:
Merge pull request #8037 from Azure/v-sabiraj-NISTworkbookchanges

Updated NIST Solution for workbook

v-dvedak committed:
Merge pull request #7953 from Accelerynt-Security/05-01-2023--AS-Edgescan-Integration

05 01 2023  as edgescan integration

v-dvedak committed:
Merge pull request #7993 from Azure/v-rusraut/UEBAEssentials

Repackaging - UEBAEssentials

v-dvedak committed:
Merge pull request #7996 from Azure/v-rbajaj/RepackagingSolutionsKQL

Repackaging Cloud Identity Threat Protection and Cloud Service Threat…

daspiker committed:
repackaged

yinghuang123 committed:
customized createSolutionV2.ps1 to build cohesity's own solution. (#19)

please hide this from public branch.

AcceleryntSecurityDev committed:
Create init

AcceleryntSecurityDev committed:
Add files via upload

AcceleryntSecurityDev committed:
Add files via upload

AcceleryntSecurityDev committed:
Delete init

AcceleryntSecurityDev committed:
Update README.md

sreedharande committed:
Fixed issues with Update Table Retention; Added support for Azure Gov

sreedharande committed:
Merge pull request #8043 from sreedharande/master

Fixed issues with Update Table Retention; Added support for Azure Gov

v-dvedak committed:
Merge pull request #7990 from daspiker/islandPackage201Update

updated package with apiVersion fix

v-dvedak committed:
Merge pull request #7623 from Flared/jct/adding_analytics_rules

Adding more Analytics rules

v-dvedak committed:
Updated API version - Cloudflare and IPQualityScore

v-dvedak committed:
Updated title of DC

v-sabiraj committed:
Updated TI connector to add Threat Intelligence Upload Indicators API

v-sabiraj committed:
removing -preview

v-sabiraj committed:
Updated version

v-dvedak committed:
Merge pull request #8048 from Azure/users/v-dvedak/fix_arm-ttk_failure

Updated API version - Cloudflare and IPQualityScore

v-dvedak committed:
Merge pull request #8050 from Azure/v-sabiraj-tiupdates

Updated TI connector to add Threat Intelligence Upload Indicators API

Commit:
Add IdentityInfo as a data connector

aprakash13 committed:
Merge pull request #8055 from Azure/ueba-data-connectors

Add IdentityInfo as a data connector

AcceleryntSecurityDev committed:
Add files via upload

yinghuang123 committed:
Merge remote-tracking branch 'origin/master' into CohesitySecurity.internal

v-dvedak committed:
Merge pull request #8041 from Accelerynt-Security/05-15-2023--AS-Add-Machine-Logon-Users-to-Incident

05 15 2023  as add machine logon users to incident

v-atulyadav committed:
Repackaged Zero nw & Armorblox

v-dvedak committed:
Merge pull request #7986 from PCNZ/patch-1

Update TenableIO.json

v-dvedak committed:
Merge pull request #8056 from Azure/v-atulyadav/zeronw

Repackaged Zero Networks & Armorblox

oshezaf committed:
Fix YAMLs

oshezaf committed:
Further yaml fixes

oshezaf committed:
More

oshezaf committed:
Merge pull request #8061 from Azure/asim/fix-schema-yamls

Fix YAMLs

yinghuang123 committed:
Merge remote-tracking branch 'origin/master' into CohesitySecurity.internal

yinghuang123 committed:
Merge remote-tracking branch 'cohesity/automation_test' into automation_test

yinghuang123 committed:
replace error_handler with set -e.
add input valiations.
and other request change in the PR review.
